packages/flutter/test/cupertino/refresh_test.dart

```diff
@@ -0,0 +1,1260 @@
+                  onRefresh: mockHelper.refreshTask,
+                ),
+                buildAListOfStuff(),
+              ],
+            ),
+          ),
+        );
+
+        final TestGesture gesture = await tester.startGesture(const Offset(0.0, 0.0));
+        // Start a refresh.
+        await gesture.moveBy(const Offset(0.0, 150.0));
+        await tester.pump();
+        verify(mockHelper.refreshTask());
+
+        // Complete the task while held down.
+        refreshCompleter.complete(null);
+        await tester.pump();
+        verify(mockHelper.builder(
+          any,
+          RefreshIndicatorMode.done,
+          150.0, // Still overscrolled here.
+          100.0, // Default value.
+          60.0, // Default value.
+        ));
+        expect(
+          tester.getRect(find.widgetWithText(Center, '0')),
+          new Rect.fromLTRB(0.0, 150.0, 800.0, 350.0),
+        );
+
+        await gesture.up();
+        await tester.pump();
+        await tester.pump(const Duration(seconds: 5));
+        expect(find.text('-1'), findsNothing);
+        expect(
+          tester.getRect(find.widgetWithText(Center, '0')),
+          new Rect.fromLTRB(0.0, 0.0, 800.0, 200.0),
+        );
+
+        debugDefaultTargetPlatformOverride = null;
+      },
+    );
+
+    testWidgets(
+      'sliver scrolled await when task completes properly removes itself',
+      (WidgetTester tester) async {
+        debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+
+        refreshIndicator = const Center(child: const Text('-1'));
+
+        await tester.pumpWidget(
+          new Directionality(
+            textDirection: TextDirection.ltr,
+            child: new CustomScrollView(
+              slivers: <Widget>[
+                new CupertinoRefreshControl(
+                  builder: mockHelper.builder,
+                  onRefresh: mockHelper.refreshTask,
+                ),
+                buildAListOfStuff(),
+              ],
+            ),
+          ),
+        );
+
+        // Start a refresh.
+        await tester.drag(find.text('0'), const Offset(0.0, 150.0));
+        await tester.pump();
+        verify(mockHelper.refreshTask());
+
+        await tester.drag(find.text('0'), const Offset(0.0, -300.0));
+        await tester.pump();
+
+        // Refresh indicator still being told to layout the same way.
+        verify(mockHelper.builder(
+          any,
+          RefreshIndicatorMode.refresh,
+          60.0,
+          100.0, // Default value.
+          60.0, // Default value.
+        ));
+
+        // Now the sliver is scrolled off screen.
+        expect(
+          tester.getTopLeft(find.widgetWithText(Center, '-1')).dy,
+          moreOrLessEquals(-175.38461538461536),
+        );
+        expect(
+          tester.getBottomLeft(find.widgetWithText(Center, '-1')).dy,
+          moreOrLessEquals(-115.38461538461536),
+        );
+
+        // Complete the task while scrolled await.
```

Comment:
scrolled await. --> scrolled.
(unless you meant something else)

Reply:
doh